src/utils/Hooks/Calculations/useAppropriateChartData.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useRef, useState } from 'react';
+import { differenceInMonths, differenceInWeeks } from 'date-fns';
+import { CategoryCodes, ChartData, MeasurementTypeCodesLabel, TimeUnitCodes } from '../../../types/chartDataTypes';
+
+export const useAppropriateChartData = (
+    chartDataForGender: ChartData,
+    dateOfBirth: Date,
+    defaultIndicator: string,
+    gender: string,
+    setDefaultIndicatorError: (value: boolean) => void,
+) => {
+    const [selectedCategory, setSelectedCategory] = useState<keyof typeof CategoryCodes>();
+    const [selectedDataset, setSelectedDataset] = useState<string>();
+
+    const selectDatasetForCategoryRef = useRef<(category: keyof typeof CategoryCodes) => void>();
+    selectDatasetForCategoryRef.current = (category: keyof typeof CategoryCodes) => {
+        const { datasets } = chartDataForGender[category];
+        const childAgeInWeeks = differenceInWeeks(new Date(), dateOfBirth);
+        const childAgeInMonths = differenceInMonths(new Date(), dateOfBirth);
+        Object.entries(datasets)
+            .some(([key, value]) => {
+                const { range } = value.metadata;
+                const xAxis = value.metadata.xAxisLabel;
+
+                if (Object.values(MeasurementTypeCodesLabel)
+                    .includes(xAxis)) {
+                    setSelectedDataset((prevDataset) => (prevDataset !== key ? key : prevDataset));
+                    return true;
+                }
+
```

Comment:
A big change, if you change the current to this


```suggestion
    selectDatasetForCategoryRef.current = (category: keyof typeof CategoryCodes) => {
        const { datasets } = chartDataForGender[category];
        const childAgeInWeeks = differenceInWeeks(new Date(), dateOfBirth);
        const childAgeInMonths = differenceInMonths(new Date(), dateOfBirth);

        const isMeasurementType = (xAxis: string) =>
            Object.values(MeasurementTypeCodesLabel).includes(xAxis);

        const isWeeksInRange = (xAxis: string) =>
            xAxis === TimeUnitCodes.weeks && childAgeInWeeks < 13;

        const isMonthsInRange = (xAxis: string, range: { start: number, end: number }) =>
            xAxis === TimeUnitCodes.months && childAgeInMonths >= range.start && childAgeInMonths < range.end;

        Object.entries(datasets).some(([key, value]) => {
            const { range } = value.metadata;
            const xAxis = value.metadata.xAxisLabel;

            if (isMeasurementType(xAxis) || isWeeksInRange(xAxis) || isMonthsInRange(xAxis, range)) {
                setSelectedDataset((prevDataset) => (prevDataset !== key ? key : prevDataset));
                return true;
            }

            return false;
        });
    };            
```

This code is easier to read, and does not repeat these lines:
``` 
setSelectedDataset((prevDataset) => (prevDataset !== key ? key : prevDataset));
                return true;
```

Reply:
The helper functions can also be placed outside of the .current, but this does not really matter, it is just normal for helper functions to be outside. It makes the .current function easier to read